database/mongodb/mongodb.go

```diff
@@ -23,6 +26,14 @@ func init() {
 
 var DefaultMigrationsCollection = "schema_migrations"
 
+const DefaultLockingCollection = "migrate_advisory_lock" // the collection to use for advisory locking by default.
+const lockKeyUniqueValue = 0                             // the unique value to lock on. If multiple clients try to insert the same key, it will fail (locked).
+const DefaultLockTimeout = 15                            // the default maximum time to wait for a lock to be released.
+const DefaultLockTimeoutInterval = 10                    // the default maximum intervals time for the locking timout.
+const DefaultAdvisoryLockingFlag = true                  // the default value for the advisory locking feature flag. Default is true.
+const LockIndexName = "lock_unique_key"                  // the name of the index which adds unique constraint to the locking_key field.
```

Comment:
Do index names need to be unique across the DB or just within a collection? The [docs](https://docs.mongodb.com/manual/reference/method/db.collection.createIndex/#collation-option) I skimmed didn't mention anything about this. If it's unique by collection, then using a `const` is fine, but if it's unique by db, then we should make this configurable.

Reply:
They are unique across Collection, not DB